corpus/single_message.json

```diff
@@ -1,29 +1,35 @@
 {
-  "class": "message",
+  "class": "messages",
   "selector": "<some predicate for the driver to determine if this vector applies to the implementation, e.g. to deal with protocol upgrades>",
-
   "_meta": {
     "id": "test_vector_id",
     "version": "version_id",
+    "description": "tests that the VM rejects a message containing...",
+    "comment": "this message should never actually be fed to the VM, because the block would be deemed invalid by the syncer",
     "gen": {
       "source": "lotus",
       "version": "0.4.1+git.27d74337+api0.8.1"
     }
   },
-
   "car_bytes": "0x890043009920583103b5b1fab6769fcb464146d192af785915d8e1dacff264b2be3bcb7bf4064a77cb3b13e27063ea4825dc48cd0a7df77e211916e748002740f0c493a2824200011a09d9c4dc0758c68219091c58c083e7c59e748e070b382a813f4ffe8ead2a090469aaa37f0664b82c092bc78b43fac300335cd2d6e4a8c71ce138003c0fa8ee1707b41aa87f35c811613d59ac53449aef8263aa51c876f2e67c185118ec9628f73442f58e093909ce952e431e770ac693cdbe99f2f9404ad3ca754196ab443c6c66afa41310e705e453496bdcedf0f271529dc7374021d7b80d343397f996d06bf11e753a49eaed7b3ce80a8e9bbaa48bf83223527bd8a2933f002949fdb8058103336a4a979375d5e963cdd7b1",
-
   "preconditions": {
-    "height": 100,
+    "epoch": 100,
     "state_tree": {
       "root_cid": {
         "/": "bafy2bzacebbxsepazfgwepawspvzenb2x64pmqjyan3wgtfpxu5nxez33wzkc"
       }
     }
   },
-
-  "apply_message": "0x89004300ba20583103b85358cc1c968d826c0a0efcde99b6e4db138edfc051f188a4471cfc23ccf7884adbbac912e45f7363774bf16f83ca8f1902b44800505e1442e66b744200011a09d225c80758c68219016f58c099a447a8c80988e1c061dd3a8dd674033c3a6083c4b55cf035e6e8de3baf09436ad00adf6eb14d58ae6691d22ce82255874047e7aa0b9a86bbf424afc6d89dc29f7b253dd04c88a625b86345f05aa0bdee503971c1e9735ed5b81c31ad41a1f10b387d96151d1cbd0c4286267daf04087fd8d5272f870994c3732f3fc4f54287a485d6c8dafb6056df3fc9233f849027b4c1272850435126fef65a8363768a858c0cc5f9d9114ff378ad50cab25ab0a2f685ca63d9f1f7727da4d06db5b80b5f",
-
+  "apply_messages": [
+    {
+      "msg": "0x89004300ba20583103b85358cc1c968d826c0a0efcde99b6e4db138edfc051f188a4471cfc23ccf7884adbbac912e45f7363774bf16f83ca8f1902b44800505e1442e66b744200011a09d225c80758c68219016f58c099a447a8c80988e1c061dd3a8dd674033c3a6083c4b55cf035e6e8de3baf09436ad00adf6eb14d58ae6691d22ce82255874047e7aa0b9a86bbf424afc6d89dc29f7b253dd04c88a625b86345f05aa0bdee503971c1e9735ed5b81c31ad41a1f10b387d96151d1cbd0c4286267daf04087fd8d5272f870994c3732f3fc4f54287a485d6c8dafb6056df3fc9233f849027b4c1272850435126fef65a8363768a858c0cc5f9d9114ff378ad50cab25ab0a2f685ca63d9f1f7727da4d06db5b80b5f",
+      "receipt": {
```

Comment:
this seems like a validation condition rather than a thing we're applying.
Probably only relevant for messages, not for other classes.
maybe add to post conditions when in `messages` class vector, and expect the length of receipts to equal the length of messages being applied.

Reply:
Good point; fixed in [`bdb3871` (#209)](https://github.com/filecoin-project/oni/pull/209/commits/bdb387120e40a0ae9e5b2f10d241f5da47fd90c4). Although JSON schema won't allow me to enforce:

- that `len(postconditions.receipts) == len(apply_messages)`.
- that when `class == messages`, `postconditions.receipts` must be present (although this is implicit in the above constraint)

Damn you, JSON Schema. I documented these constraints as descriptions; we'll need to enforce them in code.